### DISCUSSION TITLE
Only syncing events marked 'busy'

### DISCUSSION THREAD

**Discussion by peckevin-github** [derekantrican/GAS-ICS-Sync#371] [Q&A] [answered]
I've looked around for an answer for this as well but haven't found anything related - how would I go about only syncing events marked 'busy' and not syncing those marked 'free'?

Alternatively, it looks like when the script runs it isn't respecting the free / busy flag from the original Apple calendar event (it is marked 'busy' no matter what in the synced Google calendar event). If there was a way to have it respect the original free/busy that would be amazing too!

**Comment by jonas0b1011001** [comment]
I don't know how apple marks events as free/busy, can you share an example?

- open the *.ics file in a texteditor
- search for the event title
- post everything between BEGIN:VEVENT and END:VEVENT (you can remove any sensitive data of course)

**Comment by peckevin-github** [comment]
Looks like it may not mark it, which is odd. Here's what's in between the BEGIN:VEVENT and END:EVENT for an event marked 'free'

BEGIN:VEVENT
CREATED:20230928T184210Z
DTEND;TZID=America/Detroit:20231001T130000
DTSTAMP:20230928T184720Z
DTSTART;TZID=America/Detroit:20231001T120000
LAST-MODIFIED:20230928T184444Z
SEQUENCE:0
SUMMARY:Test Free Event
TRANSP:TRANSPARENT
UID:5C8ADE3B-4063-40FB-8791-78B9CF319D0B
URL;VALUE=URI:
X-APPLE-CREATOR-IDENTITY:com.apple.calendar
X-APPLE-CREATOR-TEAM-IDENTITY:0000000000
BEGIN:VALARM
ACTION:NONE
TRIGGER;VALUE=DATE-TIME:19760401T005545Z
END:VALARM
END:VEVENT

**Reply by jonas0b1011001** [reply]
Hmm, does not seem like there is any indication. Could you share a 'busy' event aswell for comparison?

**Reply by peckevin-github** [reply]
It looks like maybe it's using transparency? That's the only real difference I see: 

BEGIN:VEVENT
CREATED:20230929T125834Z
DTEND;TZID=America/Detroit:20230930T170000
DTSTAMP:20230929T125943Z
DTSTART;TZID=America/Detroit:20230930T160000
LAST-MODIFIED:20230929T125834Z
SEQUENCE:0
SUMMARY:Test Busy Event
TRANSP:OPAQUE
UID:7AFB643F-F383-42E1-88C2-C502211A1535
X-APPLE-CREATOR-IDENTITY:com.apple.calendar
X-APPLE-CREATOR-TEAM-IDENTITY:0000000000
END:VEVENT

**Reply by jonas0b1011001** [reply]
That's just like google is doing it and transparency is properly synced by the script, i just tested that on my end.

Looking at the other discussion you opened, maybe you accidentally removed syncing the _transp_ property (Helpers.gs [lines 398-402](https://github.com/derekantrican/GAS-ICS-Sync/blob/e5cd8e02417ee38a04eadbd795201822dbac2e2d/Helpers.gs#L398))?

**Reply by peckevin-github** [reply]
So if you have an event marked Free in your Apple Calendar, when it's synced to Google it's showing up free as well? I have the transparency lines still in my Helpers.gs file.

**Reply by jonas0b1011001** [reply]
I don't have an Apple Calendar to test, but syncing the events you posted properly sets them free/busy.
<img width="793" alt="Bildschirmfoto 2023-09-29 um 15 40 14" src="https://github.com/derekantrican/GAS-ICS-Sync/assets/43352574/fac0e518-8925-4c3f-b924-459715fe7ba7">
<img width="793" alt="Bildschirmfoto 2023-09-29 um 15 40 27" src="https://github.com/derekantrican/GAS-ICS-Sync/assets/43352574/bd6c4f79-0a13-42a9-a16e-de12e2c0d058">

**Reply by peckevin-github** [reply]
Hmm okay, looks like the events themselves are fine then. You've helped so much already, but would you mind trying to sync this calendar? It only has 2 events, one on October 2 marked free and one on October 3 marked busy.

https://p117-caldav.icloud.com/published/2/MTkyNzk1OTc5NjE5Mjc5NcIeA2C_xZRf2YVcLtK8RNBsEXMkb4-xwwCO8R6wQPuvTl8u9mY8lk7eOlR3PA7h9xkAyP28YRNUqLIgfw_Yu4I

**Reply by jonas0b1011001** [reply, marked as answer]
They are both 'busy'. The .ics is missing the _transp_ info.

**Reply by peckevin-github** [reply]
Thanks for checking. It must get stripped out when sharing then, which is unfortunate.
